test/controllers/power_controller_test.rb

```diff
@@ -0,0 +1,210 @@
+require 'test_helper'
+
+class PowerControllerTest < ActionDispatch::IntegrationTest
+  setup do
+    @joe = users(:joe)
+    post user_session_path, params: { 'user[email]' => @joe.email, 'user[password]' => 'passwordjoe' }
+  end
+
+  test 'joe can create project' do
+    post "#{projects_url}.json", params: { project: { name: "2nd Joe Project" } }
+    assert_response :success
+  end
+
+  test 'joe can add users to joe_project ' do
+    @joe_project = projects(:joe_project)
+    put add_user_project_url(@joe_project), params: { user_email_to_add: "abc@xyz.com" }
+    assert_response :success
+  end
+
+  test 'joe can change project visibility to joe_project ' do
+    @joe_project = projects(:joe_project)
+    put update_project_visibility_project_url(@joe_project), params: { public_access: 0 }
```

Comment:
from product perspective this is like "undo send" for email

Reply:
oh i see this is only allowed for a private project;  below you have a test to disallow it for a public project